backend/src/schema/resolvers/posts.js

```diff
@@ -29,16 +28,31 @@ const filterForBlockedUsers = async (params, context) => {
   return params
 }
 
+const maintainPinnedPosts = params => {
+  const pinnedPostFilter = { pinnedBy_in: { role_in: ['admin'] } }
+  if (isEmpty(params.filter)) {
+    params.filter = { OR: [pinnedPostFilter, {}] }
+  } else {
+    params.filter = { OR: [pinnedPostFilter, { ...params.filter }] }
+  }
+  return params
+}
+
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-10-17T14:59:14Z" title="Thursday, October 17th 2019, 4:59:14 pm +02:00">Oct 17, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -29,10 +28,26 @@ const filterForBlockedUsers = async (params, context) => {
   return params
 }
 
+const maintainPinnedPosts = params => {
+  const pinnedPostFilter = { pinnedBy_in: { role_in: ['admin'] } }
+  if (isEmpty(params.filter)) {
+    params.filter = { OR: [pinnedPostFilter, {}] }
+  } else {
+    const filteredPostsArray = []
+    Object.keys(params.filter).forEach(key => {
+      filteredPostsArray.push({ [key]: params.filter[key] })
+    })
+    filteredPostsArray.push(pinnedPostFilter)
+    params.filter = { OR: filteredPostsArray }
+  }
+  return params
+}
+
```


Let's remove this and keep things simple stupid

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-10-17T21:50:44Z" title="Thursday, October 17th 2019, 11:50:44 pm +02:00">Oct 17, 2019</time>_

---


tried to, but couldn't get it to work the other way